undefined reference to 'fmaf' on Ubuntu 18.04

The CI builds that verify curl+rustls has started to fail on Ubuntu 18.04 and I can reproduce in a manual build as well on Ubuntu 18.04.I'm not sure exactly when this problem started but it feels like within the last few weeks. (I've tried work-arounds to get it working again in https://github.com/curl/curl/pull/7701 but so far it has failed)

This reproduces on master as well as v0.7.1 and v0.7.2.

It does not reproduce on my Debian Linux unstable.

The steps to reproduce:
~~~
cd $HOME
git clone https://github.com/rustls/rustls-ffi.git
curl https://sh.rustup.rs -sSf | sh -s -- -y
source $HOME/.cargo/env
cargo install cbindgen
cd $HOME/rustls-ffi
make
make DESTDIR=$HOME/crust install

# now build curl to use this rustls
cd $HOME
git clone https://github.com/curl/curl.git
cd curl
./buildconf
./configure --with-rustls=$HOME/crust
~~~

That configure script tries use the function `rustls_connection_read` when linking with crustls.a - but fails since it lacks the symbols `fmaf`and `fma`. Here's a piece of the resulting config.log:
~~~
configure:26660: checking for rustls_connection_read in -lcrustls
configure:26682: gcc -o conftest -Werror-implicit-function-declaration -g -O0 -pedantic -Wall -W -Wpointer-arith -Wwrite-strings -Wunused -Wshadow -Winline -Wnested-externs -Wmissing-declarations -Wmissing-prototypes -Wno-long-long -Wbad-function-cast -Wfloat-equal -Wno-multichar -Wsign-compare -Wundef -Wno-format-nonliteral -Wendif-labels -Wstrict-prototypes -Wdeclaration-after-statement -Wold-style-definition -Wstrict-aliasing=3 -Wcast-align -Wtype-limits -Wold-style-declaration -Wmissing-parameter-type -Wempty-body -Wclobbered -Wignored-qualifiers -Wconversion -Wno-sign-conversion -Wvla -ftree-vrp -Wdouble-promotion -Wformat=2 -Warray-bounds=2 -Wshift-negative-value -Wshift-overflow=2 -Wnull-dereference -fdelete-null-pointer-checks -Wduplicated-cond -Wunused-const-variable -Wduplicated-branches -Wrestrict -Walloc-zero -Wformat-overflow=2 -Wformat-truncation=2 -Wimplicit-fallthrough=4 -Wno-system-headers    -I/home/zuul/crust/include -lm   -L/home/zuul/crust/lib conftest.c -lcrustls -lpthread -ldl -lzstd  -lbrotlidec   >&5
conftest.c:46:1: warning: function declaration isn't a prototype [-Wstrict-prototypes]
 char rustls_connection_read ();
 ^~~~
/home/zuul/crust/lib/libcrustls.a(std-008055cc7d873802.std.cf1c8f7e-cgu.0.rcgu.o): In function `std::f32::<impl f32>::lerp':
/rustc/c8dfcfe046a7680554bf4eb612bad840e7631c4b//library/std/src/f32.rs:914: undefined reference to `fmaf'
/home/zuul/crust/lib/libcrustls.a(std-008055cc7d873802.std.cf1c8f7e-cgu.0.rcgu.o): In function `std::f64::<impl f64>::lerp':
/rustc/c8dfcfe046a7680554bf4eb612bad840e7631c4b//library/std/src/f64.rs:916: undefined reference to `fma'
collect2: error: ld returned 1 exit status
configure:26682: $? = 1
configure: failed program was:
| /* confdefs.h */
| #define PACKAGE_NAME "curl"
| #define PACKAGE_TARNAME "curl"
| #define PACKAGE_VERSION "-"
| #define PACKAGE_STRING "curl -"
| #define PACKAGE_BUGREPORT "a suitable curl mailing list: https://curl.se/mail/"
| #define PACKAGE_URL ""
| #define DEBUGBUILD 1
| #define CURLDEBUG 1
| #define PACKAGE "curl"
| #define VERSION "-"
| #define OS "x86_64-pc-linux-gnu"
| #define STDC_HEADERS 1
| #define HAVE_SYS_TYPES_H 1
| #define HAVE_SYS_STAT_H 1
| #define HAVE_STDLIB_H 1
| #define HAVE_STRING_H 1
| #define HAVE_MEMORY_H 1
| #define HAVE_STRINGS_H 1
| #define HAVE_INTTYPES_H 1
| #define HAVE_STDINT_H 1
| #define HAVE_UNISTD_H 1
| #define HAVE_DLFCN_H 1
| #define LT_OBJDIR ".libs/"
| #define HAVE_LDAP_SSL 1
| #define HAVE_SYS_TYPES_H 1
| #define HAVE_SYS_TIME_H 1
| #define HAVE_CLOCK_GETTIME_MONOTONIC 1
| #define HAVE_LIBBROTLIDEC 1
| #define HAVE_BROTLI_DECODE_H 1
| #define HAVE_BROTLI 1
| #define HAVE_LIBZSTD 1
| #define HAVE_ZSTD_H 1
| #define HAVE_ZSTD 1
| #define CURL_DISABLE_LDAP 1
| #define CURL_DISABLE_LDAPS 1
| #define ENABLE_IPV6 1
| #define HAVE_SOCKADDR_IN6_SIN6_SCOPE_ID 1
| #define HAVE_WRITABLE_ARGV 1
| /* end confdefs.h.  */
| 
| 
| #ifdef __cplusplus
| extern "C"
| #endif
| char rustls_connection_read ();
| int main (void)
| {
| return rustls_connection_read ();
|  ;
|  return 0;
| }
configure:26691: result: no
configure:26706: error: --with-rustls was specified but could not find rustls.

--- a/README.md
+++ b/README.md
@@ -23,6 +23,19 @@ To build in debug mode:
 
     make PROFILE=debug
 
+To link against the resulting library, on **Linux**:
+
+    -lrustls -lgcc_s -lutil -lrt -lpthread -lm -ldl -lc
+
+To link against the resulting library, on **macOS**:
+
+    -lrustls -framework Security -liconv -lSystem -lresolv -lc -lm -liconv
+
+If the linking instructions above go out of date, [you can get an up-to-date list
+via](https://doc.rust-lang.org/rustc/command-line-arguments.html#--print-print-compiler-information):
+
+    RUSTFLAGS="--print native-static-libs" cargo build
+
 # Overview
 
 Rustls doesn't do any I/O on its own. It provides the protocol handling, and

